testing/opae-c/test_init_c.cpp

```diff
@@ -166,3 +172,245 @@ TEST(init, log_file) {
+    std::string cfg_dir = (inst_cfg_dir? inst_cfg_dir : "");
+    inst_cfg_file_ = cfg_dir + std::string(inst_cfg_file) + std::string(".backup");
+    rename(OPAE_ASE_CFG_INST_PATH, inst_cfg_file_.c_str());
+
+    // find_ase_cfg at release directory
+    cfg_file = find_ase_cfg();
+    EXPECT_NE(cfg_file, nullptr);
+    rename(inst_cfg_file_.c_str(), OPAE_ASE_CFG_INST_PATH);
+}
+
+/**
+ * @test       find_ase_cfg_3
+ * @brief      Test: find_ase_cfg at HOME location
+ * @details    After renaming a configuration file located in the OPAE_ASE_CFG_SRC_PATH,
+ *             OPAE_ASE_CFG_INST_PATH and OPAE release directory<br>
+ *             When I call find_ase_cfg
+ *             Then the call is successful<br>
+ */
+TEST_P(init_ase_cfg_p, find_ase_cfg_3) {
+    char *cfg_file = nullptr;
+    char *opae_path;
+    std::string  inst_cfg_file_, opae_cfg_file_;
+    std::string  rel_cfg_file_, rel_cfg_file2_;
+
+    // copy it to a temporary buffer that we can use dirname with
+    std::string inst_cfg_path = (OPAE_ASE_CFG_INST_PATH? OPAE_ASE_CFG_INST_PATH : "");
+    std::copy(inst_cfg_path.begin(), inst_cfg_path.end(), &buffer_[0]);
+    char *inst_cfg_dir = dirname(buffer_);
+    char *inst_cfg_file = basename(buffer_);
+
+    std::string cfg_dir = (inst_cfg_dir? inst_cfg_dir : "");
+    inst_cfg_file_ = cfg_dir + std::string(inst_cfg_file) + std::string(".backup");
+    rename(OPAE_ASE_CFG_INST_PATH, inst_cfg_file_.c_str());
+
+    // rename opae_ase.cfg under releae directory
+    opae_path = getenv("OPAE_PLATFORM_ROOT");
+    std::string rel_cfg_path = (opae_path? opae_path: "");
+    rel_cfg_file_ = rel_cfg_path + std::string("/share/opae/ase/opae_ase.cfg");
+    rel_cfg_file2_ = rel_cfg_path + std::string("/share/opae/ase/opae_ase.cfg.backup");
+    rename(rel_cfg_file_.c_str(), rel_cfg_file2_.c_str());
+
+    // find_ase_cfg at HOME directory
+    cfg_file = find_ase_cfg();
+    EXPECT_NE(cfg_file, nullptr);
+
+    rename(rel_cfg_file2_.c_str(), rel_cfg_file_.c_str());  // restore file
+    rename(inst_cfg_file_.c_str(), OPAE_ASE_CFG_INST_PATH);  // restore file
+}
+
+/**
+ * @test       find_ase_cfg_4
+ * @brief      Test: find_ase_cfg at OPAE_PLATFORM_ROOT release location
+ * @details    After renaming a configuration file located in the OPAE_ASE_CFG_SRC_PATH,
+ *             OPAE_ASE_CFG_INST_PATH, OPAE release and HOME directories<br>
+ *             When I call find_ase_cfg
+ *             Then the call is successful<br>
+ */
+TEST_P(init_ase_cfg_p, find_ase_cfg_4) {
+    char *cfg_file = nullptr;
+    char *opae_path;
+    std::string  inst_cfg_file_, opae_cfg_file_;
+    std::string  rel_cfg_file_, rel_cfg_file2_;
+
+    // copy it to a temporary buffer that we can use dirname with
+    std::string inst_cfg_path = (OPAE_ASE_CFG_INST_PATH? OPAE_ASE_CFG_INST_PATH : "");
+    std::copy(inst_cfg_path.begin(), inst_cfg_path.end(), &buffer_[0]);
+    char *inst_cfg_dir = dirname(buffer_);
+    char *inst_cfg_file = basename(buffer_);
```

Comment:
Should verify these before referencing them.

Reply:
Will do the checking.

Reply:
Will need to use strcmp..